cl/search/tests.py

```diff
@@ -3670,3 +3670,91 @@ def test_oa_results_relevance_ordering_elastic(self) -> None:
             < r.content.decode().index("Jose"),
             msg="'Hong Liu Yang' should come BEFORE 'Hong Liu Lorem' and 'Jose' when order_by relevance.",
         )
+
+    def test_oa_results_api_fields_es(self) -> None:
+        search_params = {
+            "type": SEARCH_TYPES.ORAL_ARGUMENT,
+            "q": "Hong Liu Lorem v. Lynch-Loretta E.",
+        }
+
+        # API
+        r = self.client.get(
+            reverse("search-list", kwargs={"version": "v3"}), search_params
+        )
+        actual = self.get_results_count(r)
+        expected = 1
+        self.assertEqual(actual, expected)
+
+        keys_to_check = [
+            "absolute_url",
+            "caseName",
+            "court",
+            "court_citation_string",
+            "court_exact",
+            "court_id",
+            "dateArgued",
+            "dateReargued",
+            "dateReargumentDenied",
+            "docketNumber",
+            "docket_id",
+            "download_url",
+            "duration",
+            "file_size_mp3",
+            "id",
+            "judge",
+            "local_path",
+            "panel_ids",
+            "snippet",
+            "source",
+            "timestamp",
+        ]
+        keys_count = len(r.data["results"][0])
+        self.assertEqual(keys_count, 21)
+        for key in keys_to_check:
+            if key in r.data["results"][0]:
+                key_is_present = True
+            else:
+                key_is_present = None
+            self.assertIsNotNone(
+                key_is_present,
+                msg=f"Key {key} not found in the result object.",
+            )
+
+    def test_oa_results_api_pagination(self) -> None:
+        with transaction.atomic():
+            for i in range(20):
+                AudioFactory.create(
+                    docket_id=self.audio_3.docket.pk,
+                )
+            self.rebuild_index()
+            search_params = {
+                "type": SEARCH_TYPES.ORAL_ARGUMENT,
+            }
+            # API
+            r = self.client.get(
+                reverse("search-list", kwargs={"version": "v3"}), search_params
+            )
+            actual = self.get_results_count(r)
+            expected = 20
+            self.assertEqual(actual, expected)
+            self.assertEqual(24, r.data["count"])
+            self.assertIn("page=2", r.data["next"])
+
+            # Test next page.
+            search_params = {
+                "type": SEARCH_TYPES.ORAL_ARGUMENT,
+                "page": 2,
+            }
+            r = self.client.get(
+                reverse("search-list", kwargs={"version": "v3"}), search_params
+            )
+            actual = self.get_results_count(r)
+            expected = 4
+            self.assertEqual(actual, expected)
+            self.assertEqual(24, r.data["count"])
+            self.assertEqual(None, r.data["next"])
+
+            # Remove Audio objects to avoid affecting other concurrent tests.
```

Comment:
I thought the `ESTestCaseMixin` meant they're not concurrent? I'd love it if they were, but...

Reply:
Yeah, they're not concurrent. But sometimes the order of tests could change?  Even, if this is the last test in this class if the objects created in this test are not removed from the index, it affects other tests that expect fewer elements in the index, as described in a comment above.